WalletWasabi/Crypto/Api/WabiSabiException.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Runtime.Serialization;
+
+namespace WalletWasabi.Crypto.Api
+{
+	public enum WabiSabiErrorCode
+	{
+		Unspecified = 0,
+		SerialNumberAlreadyUsed = 1,
+		CoordinatorReceivedInvalidProofs = 2,
+		NegativeBalance = 3,
+		InvalidBitCommitment = 4,
+		ClientReceivedInvalidProofs = 5,
+		IssuedCredentialNumberMismatch = 6,
+		SerialNumberDuplicated = 7,
+	}
```

Comment:
we need to consider the implications of error codes like this for security, similar to padding oracle attacks. i don't think there's a problem, but something to think about